chord_metadata_service/mcode/models.py

```diff
@@ -0,0 +1,276 @@
+    """
+    Class  to record tests performed on patient.
+    """
+    # TODO Should this class be a part of Patients app? patient related metadata
+    id = models.CharField(primary_key=True, max_length=200,
+                          help_text=rec_help(d.LABS_VITAL, "id"))
+    individual = models.ForeignKey(Individual, on_delete=models.CASCADE,
+                                   help_text=rec_help(d.LABS_VITAL, "individual"))
+    body_height = JSONField(validators=[quantity_validator], help_text=rec_help(d.LABS_VITAL, "body_height"))
+    body_weight = JSONField(validators=[quantity_validator], help_text=rec_help(d.LABS_VITAL, "body_weight"))
+    # corresponds to DiagnosticReport.result - complex element, probably should be changed to Array of json
+    cbc_with_auto_differential_panel = ArrayField(models.CharField(max_length=200), blank=True, null=True,
+                                                  help_text=rec_help(d.LABS_VITAL, "cbc_with_auto_differential_panel"))
+    comprehensive_metabolic_2000 = ArrayField(models.CharField(max_length=200), blank=True, null=True,
+                                              help_text=rec_help(d.LABS_VITAL, "comprehensive_metabolic_2000"))
+    blood_pressure_diastolic = JSONField(blank=True, null=True, validators=[quantity_validator],
+                                         help_text=rec_help(d.LABS_VITAL, "blood_pressure_diastolic"))
+    blood_pressure_systolic = JSONField(blank=True, null=True, validators=[quantity_validator],
+                                        help_text=rec_help(d.LABS_VITAL, "blood_pressure_systolic"))
+    #TODO Change CodeableConcept to Ontology class
+    tumor_marker_test = JSONField(validators=[tumor_marker_test_validator],
+                                  help_text=rec_help(d.LABS_VITAL, "tumor_marker_test"))
+
+    def __str__(self):
+        return str(self.id)
+
+    def clean(self):
+        if not (self.blood_pressure_diastolic or self.blood_pressure_systolic):
+            raise ValidationError('At least one of the following must be reported: Systolic Blood Pressure or'
+                                  'Diastolic Blood Pressure.')
+
+
+################################# Disease #################################
+
+class CancerCondition(models.Model, IndexableMixin):
+    """
+    Class to record the history of primary or secondary cancer conditions.
+    """
+    CANCER_CONDITION_TYPE = (
+        ('primary', 'primary'),
+        ('secondary', 'secondary')
+    )
+    id = models.CharField(primary_key=True, max_length=200, help_text=rec_help(d.CANCER_CONDITION, "id"))
+    condition_type = models.CharField(choices=CANCER_CONDITION_TYPE, max_length=200,
+                                      help_text=rec_help(d.CANCER_CONDITION, "condition_type"))
+    # TODO add body_location_code validator array of json
+    body_location_code = ArrayField(JSONField(null=True, blank=True, validators=[ontology_class_validator]),
```

Comment:
This should be consistent with what I did in the experiments model, where we picked a JSONField containing an array rather than an array of JSONFields. We can review the decision but it should be consistent across models.